### PR TITLE
Feature/condo/doma 1501/replace debug message func to pino logger

### PR DESCRIPTION
**Pull request by AntonAL**
Replaced `console.log` to Pino JSON logger, used in Keystone

Used namespace `sbbol/subscriptions` namespace for Fintech API interaction.

- [x] Replace `console.log` in `SbbolOauth2Api`

### CODE REVIEW

File: apps/condo/domains/organization/integrations/sbbol/sync/syncPaymentRequestsForSubscriptions.js [hunk 1]
```diff
@@ -6,28 +6,27 @@ const { SbbolFintechApi } = require('../SbbolFintechApi')
...
     const { keystone: context } = await getSchemaCtx('ServiceSubscriptionPayment')
-    debugMessage('Start postPaymentRequestsFor', subscription)
-    debugMessage(`Processing ServiceSubscription(type="sbbol", id="${subscription.id}")`)
+    logger.info({ message: 'Start postPaymentRequestsFor', subscription })
 
     if (!subscription.sbbolOfferAccept) {
-        console.error(`${SBBOL_OFFER_ACCEPT_IS_NULL_ERROR} Cannot obtain requisites for payer from latest ServiceSubscription`)
-        debugMessage('Abort postPaymentRequestsFor')
+        logger.error({ message: `${SBBOL_OFFER_ACCEPT_IS_NULL_ERROR} Cannot obtain requisites for payer from latest ServiceSubscription`, subscription })
+        logger.info({ message: 'Abort postPaymentRequestsFor', subscription })
```
Comment (pahaz) [inline apps/condo/domains/organization/integrations/sbbol/sync/syncPaymentRequestsForSubscriptions.js:21]: it's better to leave one logger message

File: apps/condo/domains/organization/integrations/sbbol/sync/syncPaymentRequestsForSubscriptions.js [hunk 2]
```diff
@@ -6,28 +6,27 @@ const { SbbolFintechApi } = require('../SbbolFintechApi')
...
         return
     }
 
     const { payerAccount, payerBankBic, payerBankCorrAccount, payerInn, payerName } = subscription.sbbolOfferAccept
 
     if (!payerAccount || !payerBankBic || !payerBankCorrAccount || !payerInn || !payerName) {
-        console.error(SBBOL_OFFER_ACCEPT_HAS_INCORRECT_PAYER_REQUISITES_ERROR, subscription.sbbolOfferAccept)
-        debugMessage('Abort postPaymentRequestsFor')
+        logger.error({ message: SBBOL_OFFER_ACCEPT_HAS_INCORRECT_PAYER_REQUISITES_ERROR, sbbolOfferAccept: subscription.sbbolOfferAccept })
```
Comment (pahaz) [inline apps/condo/domains/organization/integrations/sbbol/sync/syncPaymentRequestsForSubscriptions.js:28]: leave one message with all context

File: apps/condo/domains/organization/integrations/sbbol/sync/syncSbbolSubscriptionPaymentRequestsState.js [hunk 3]
```diff
@@ -16,16 +16,16 @@ const syncSbbolSubscriptionPaymentRequestStateFor = async (payment, fintechApi)
...
         const { keystone: context } = await getSchemaCtx('ServiceSubscriptionPayment')
         if (!data.bankStatus) {
-            console.error('Error: Status response does not contains "bankStatus" field. Something was changed in SBBOL Fintech API.')
+            logger.error({ message: 'Error: Status response does not contains "bankStatus" field. Something was changed in SBBOL Fintech API.' })
         } else {
             if (get(payment.statusMeta, 'bankStatus') !== data.bankStatus) {
                 const status = SBBOL_PAYMENT_STATUS_MAP[data.bankStatus]
                 if (!status) {
-                    console.error(`Error: status "${data.bankStatus}" cannot be mapped to status values of ServiceSubscriptionPayment, because it does not belongs to the list of known statuses. Consider to add it.`)
+                    logger.error({ message: `Error: status "${data.bankStatus}" cannot be mapped to status values of ServiceSubscriptionPayment, because it does not belongs to the list of known statuses. Consider to add it.` })
```
Comment (pahaz) [inline apps/condo/domains/organization/integrations/sbbol/sync/syncSbbolSubscriptionPaymentRequestsState.js:28]: it's better to extract status as another context value

File: apps/condo/domains/organization/integrations/sbbol/sync/syncSubscriptions.js [hunk 4]
```diff
@@ -65,7 +65,7 @@ const syncSubscriptionsFor = async (advanceAcceptance) => {
     if (active) {
         // TODO: add trial for additional day when client accepts previously revoked (after accepting) offer
 
-        debugMessage(`User from organization(inn=${payerInn}) has accepted our offer in SBBOL`)
+        logger.info({ message: `User from organization(inn=${payerInn}) has accepted our offer in SBBOL`, payerInn })
```
Comment (pahaz) [inline apps/condo/domains/organization/integrations/sbbol/sync/syncSubscriptions.js:68]: extract inn

File: apps/condo/domains/organization/integrations/sbbol/sync/syncSubscriptions.js [hunk 5]
```diff
@@ -89,9 +89,9 @@ const syncSubscriptionsFor = async (advanceAcceptance) => {
                 ...advanceAcceptance, // TODO: Figure out, why it crashes here on `payerInn` field
             },
         })
-        debugMessage('Created trial subscription for SBBOL', trialServiceSubscription)
+        logger.info({ message: 'Created trial subscription for SBBOL', trialServiceSubscription })
     } else {
-        debugMessage(`User from organization(inn=${payerInn}) has declined our offer in SBBOL`)
+        logger.info({ message: `User from organization(inn=${payerInn}) has declined our offer in SBBOL`, payerInn })
```
Comment (pahaz) [inline apps/condo/domains/organization/integrations/sbbol/sync/syncSubscriptions.js:94]: it's better to remove context from the message (inn)

File: apps/condo/domains/organization/integrations/sbbol/sync/syncPaymentRequestsForSubscriptions.js [hunk 6]
```diff
@@ -6,28 +6,27 @@ const { SbbolFintechApi } = require('../SbbolFintechApi')
...
 const conf = process.env
 const SBBOL_CONFIG = conf.SBBOL_CONFIG ? JSON.parse(conf.SBBOL_CONFIG) : {}
-const { debugMessage } = require('../common')
+const { logger } = require('../common')
 const { dvSenderFields, BANK_OPERATION_CODE } = require('../constants')
 const { SBBOL_OFFER_ACCEPT_IS_NULL_ERROR, SBBOL_OFFER_ACCEPT_HAS_INCORRECT_PAYER_REQUISITES_ERROR } = require('@condo/domains/subscription/constants/errors')
 
 // TODO(antonal): Add 3 days for trial subscription when payment is emitted to compensate payment processing lag and subsequent in service unavailability
 
 const postPaymentRequestFor = async (subscription, fintechApi) => {
```
Comment (pahaz) [inline apps/condo/domains/organization/integrations/sbbol/sync/syncPaymentRequestsForSubscriptions.js:15]: it's better to include more context inside logger messages like inn payer name .. and so on

File: apps/condo/domains/organization/integrations/sbbol/sync/syncPaymentRequestsForSubscriptions.js (hunk 1 above)
Comment (AntonAL) [reply apps/condo/domains/organization/integrations/sbbol/sync/syncPaymentRequestsForSubscriptions.js:21]: Changed

File: apps/condo/domains/organization/integrations/sbbol/sync/syncPaymentRequestsForSubscriptions.js (hunk 2 above)
Comment (AntonAL) [reply apps/condo/domains/organization/integrations/sbbol/sync/syncPaymentRequestsForSubscriptions.js:28]: Changed

File: apps/condo/domains/organization/integrations/sbbol/sync/syncSbbolSubscriptionPaymentRequestsState.js (hunk 3 above)
Comment (AntonAL) [reply apps/condo/domains/organization/integrations/sbbol/sync/syncSbbolSubscriptionPaymentRequestsState.js]: Changed

File: apps/condo/domains/organization/integrations/sbbol/sync/syncPaymentRequestsForSubscriptions.js (hunk 6 above)
Comment (AntonAL) [reply apps/condo/domains/organization/integrations/sbbol/sync/syncPaymentRequestsForSubscriptions.js:18]: Included more context.

File: apps/condo/domains/organization/integrations/sbbol/sync/syncSubscriptions.js (hunk 5 above)
Comment (AntonAL) [reply apps/condo/domains/organization/integrations/sbbol/sync/syncSubscriptions.js:94]: Is it private data?

Comment (AntonAL) [reply apps/condo/domains/organization/integrations/sbbol/sync/syncSubscriptions.js]: Removed

File: apps/condo/domains/organization/integrations/sbbol/oauth2.js [hunk 7]
```diff
@@ -136,22 +137,38 @@ class SbbolOauth2Api {
...
-                        console.log('--> %s %s', options.method.toUpperCase(), options.url.href)
-                        console.log('--> HEADERS %o', options.headers)
+                        const logData = {
+                            method: options.method.toUpperCase(),
+                            url: options.url.href,
+                            headers: options.headers,
+                        }
                         if (options.body) {
-                            console.log('--> BODY %s', options.body)
+                            console.debug(options.body)
```
Comment (pahaz) [inline apps/condo/domains/organization/integrations/sbbol/oauth2.js:146]: de we still need it?

File: apps/condo/domains/organization/integrations/sbbol/oauth2.js [hunk 8]
```diff
@@ -136,22 +137,38 @@ class SbbolOauth2Api {
...
-                        console.log('<-- HEADERS %o', response.headers)
+                        const logData = {
+                            statusCode: response.statusCode,
+                            method: response.request.options.method.toUpperCase(),
+                            url: response.request.options.url.href,
+                            headers: response.headers,
+                        }
+                        console.log()
                         if (response.body) {
                             console.log('<-- BODY %s', response.body)
```
Comment (pahaz) [inline apps/condo/domains/organization/integrations/sbbol/oauth2.js:165]: do we still need it?

File: apps/condo/domains/organization/integrations/sbbol/sync/syncPaymentRequestsForSubscriptions.js [hunk 9]
```diff
@@ -6,46 +6,63 @@ const { SbbolFintechApi } = require('../SbbolFintechApi')
...
 
     if (!payerAccount || !payerBankBic || !payerBankCorrAccount || !payerInn || !payerName) {
-        console.error(SBBOL_OFFER_ACCEPT_HAS_INCORRECT_PAYER_REQUISITES_ERROR, subscription.sbbolOfferAccept)
-        debugMessage('Abort postPaymentRequestsFor')
+        logger.error({
+            message: 'Latest ServiceSubscription has insufficient data in offer accept',
+            error: SBBOL_OFFER_ACCEPT_HAS_INCORRECT_PAYER_REQUISITES_ERROR,
+            sbbolOfferAccept: subscription.sbbolOfferAccept,
+        })
+        logger.info({ message: 'Abort postPaymentRequestsFor', subscription })
```
Comment (pahaz) [inline apps/condo/domains/organization/integrations/sbbol/sync/syncPaymentRequestsForSubscriptions.js:38]: it's better to remove it and add the subscription context inside the error line ^^

File: apps/condo/domains/organization/integrations/sbbol/sync/syncSbbolSubscriptionPaymentRequestsState.js [hunk 10]
```diff
@@ -12,20 +12,26 @@ const dayjs = require('dayjs')
...
-            console.error('Error: Status response does not contains "bankStatus" field. Something was changed in SBBOL Fintech API.')
+            logger.error({
+                message: 'Status response does not contains "bankStatus" field. Maybe, something was changed in setup of SBBOL Fintech API or in API itself.',
+            })
         } else {
             if (get(payment.statusMeta, 'bankStatus') !== data.bankStatus) {
                 const status = SBBOL_PAYMENT_STATUS_MAP[data.bankStatus]
                 if (!status) {
-                    console.error(`Error: status "${data.bankStatus}" cannot be mapped to status values of ServiceSubscriptionPayment, because it does not belongs to the list of known statuses. Consider to add it.`)
+                    logger.error({
```
Comment (pahaz) [inline apps/condo/domains/organization/integrations/sbbol/sync/syncSbbolSubscriptionPaymentRequestsState.js:31]: do you need `payment` here?

File: apps/condo/domains/organization/integrations/sbbol/sync/syncSubscriptions.js [hunk 11]
```diff
@@ -108,29 +110,34 @@ const syncSubscriptionsFor = async (advanceAcceptance) => {
...
         // `service_organization_hashOrgId` is a `userInfo.HashOrgId` from SBBOL, that used to obtain accessToken
         // for organization, that will be queried in SBBOL using `SbbolFintechApi`.
         ourOrganizationAccessToken = await SbbolRequestApi.getOrganizationAccessToken(SBBOL_CONFIG.service_organization_hashOrgId)
     } catch (e) {
-        console.error('syncSubscriptions() Error:', e)
+        logger.error({
+            message: 'Failed to obtain organization access token from SBBOL',
+            error: e.message,
+            hashOrgId: SBBOL_CONFIG.service_organization_hashOrgId,
+            module,
```
Comment (pahaz) [inline apps/condo/domains/organization/integrations/sbbol/sync/syncSubscriptions.js:125]: that is this? module?

File: apps/condo/domains/organization/integrations/sbbol/oauth2.js (hunk 7 above)
Comment (AntonAL) [reply apps/condo/domains/organization/integrations/sbbol/oauth2.js:146]: Oops, it's redundant

File: apps/condo/domains/organization/integrations/sbbol/oauth2.js (hunk 8 above)
Comment (AntonAL) [reply apps/condo/domains/organization/integrations/sbbol/oauth2.js:165]: Oops, it's redundant

**Review by pahaz — APPROVED** [review]

**Pull request merged**